develop/devguide/Automation/Howto/Creating_a_cypress_test_for_an_interactive_automation_script.md

```diff
@@ -0,0 +1,219 @@
+             throw;
+          }
+          catch (InteractiveUserDetachedException)
+          {
+             // Catch a user detaching from the interactive script by closing the window.
+             // Only applicable for interactive scripts, can be removed for non-interactive scripts.
+             throw;
+          }
+          catch (Exception e)
+          {
+             engine.ExitFail($"Run|Something went wrong: {e}");
+          }
+       }
+   }
+
+   public class HelloWorldDialog : Dialog
+   {
+      public HelloWorldDialog(IEngine engine) : base(engine)
+      {
+          // Create a label widget
+          var label = new Label("Hello, World!") { Style = TextStyle.Title };
+          
+          // Define the debug tag to be used in the cypress test
+          label.DebugTag = "label";
+          
+          // Add it to the dialog grid at position 0,0
+          AddWidget(label, 0, 0);
+          
+          // Create a button widget
+          var button = new Button("OK");
+          
+          // Define the debug tag to be used in the cypress test
+          button.DebugTag = "button";
+          
+          // Add it to the dialog just below the label
+          AddWidget(button, 1, 0);
+          
+          // Add an action to be performed whenever the button is clicked
+          button.Pressed += (sender, args) => engine.ExitSuccess("Done");
+      }
+    }
+    ```
+
+## Setting up the web environment
+
+1. [Create a new low-code app](xref:Creating_custom_apps).
+
+1. Add a button to the low-code app and configure it to trigger the created interactive Automation script when clicked.
+
+   For more information, see [Button](xref:DashboardButton).
+
+## Creating the Cypress test
+
+1. In the `cypress/e2e` folder, create a test file with a `.cy.js` extension, such as `HelloWorldDialog.cy.js`.
+
+1. Implement the Cypress test below to log in, navigate to the low-code app, and interact with the Automation script.
+
+   > [!NOTE]
+   > Ensure all variable values in the test below are updated with the correct information (*dmaAddress*, *username*, *password*, etc.) before you run the test. This allows the script to interact properly with your specific application environment.
+
+   ```javascript
+   describe('Test Interactive Automation Script', () => {
+       // Define the necessary variables and replace with your specific details
+       const dmaAddress = ''; // Replace with the IP or hostname of a DMA (e.g. 'localhost')
+       const username = ''; // Replace with the user username
+       const password = ''; // Replace with the user password
+       const buttonLabel = ''; // Replace with the button label
+       const appId = ''; // Replace with the app ID
+
+       // Before running the tests, visit the low-code app
+       before(() => cy.visit(`http://${dmaAddress}/app/${appId}`));
+
+       it('should log in and execute the Interactive Automation Script', () => {
```

Comment:
S: 'should log in, execute the Interactive Automation Script, check label and click the button'